pages/dao/[symbol]/proposal/components/instructions/raydium/AddLiquidity.tsx

```diff
@@ -111,27 +103,23 @@ const AddLiquidityRaydium = ({
       form.governedAccount?.governance?.account &&
       wallet?.publicKey
     ) {
-      const tokenA = getGovernanceToken(connection.cluster, form.tokenAName)
-      const tokenB = getGovernanceToken(connection.cluster, form.tokenBName)
-
+      const poolKeys = getLiquidityPoolKeysByLabel(form.liquidityPool)
+      const base = await connection.current.getTokenSupply(poolKeys.baseMint)
+      const quote = await connection.current.getTokenSupply(poolKeys.quoteMint)
       const createIx = createAddLiquidityInstruction(
-        new PublicKey(tokenA.address),
-        new PublicKey(tokenB.address),
+        poolKeys,
         new BN(
-          (
-            Number(Number(form.tokenAAmountIn).toFixed(tokenA.decimals)) *
-            10 ** tokenA.decimals
-          ).toString()
+          new BigNumber(form.baseAmountIn.toString())
+            .shiftedBy(-base.value.decimals.toString())
```

Comment:
`shiftedBy` takes a number in. I don't think that you need `.toString()`

```
.shiftedBy(-quote.value.decimals)
```

Reply:
fixed